tests/utils/stl/test/format.py

```diff
@@ -254,8 +254,11 @@ def getBuildSteps(self, test, lit_config, shared):
 
     def getTestSteps(self, test, lit_config, shared):
         if shared.exec_file is not None:
+            exec_env = test.cxx.compile_env
+            exec_env['TMP'] = str(shared.exec_dir)

```

Comment:
Should we also set `TEMP` and `TMPDIR` and `TEMPDIR` for maximum paranoia?

Reply:
The _tempnam documentation states that it only respects `TMP`.

Reply:
> This change is pretty targeted to that only.

Yes, I know - I'm asking if you think it's worth widening the scope a bit to preempt similar issues with future tests. I'm not sure *why* I'm asking, since it seems obvious that it would be better to not write such tests in the first place.

Nevermind.